packages/grid/x-data-grid/src/models/props/DataGridProps.ts

```diff
@@ -304,7 +304,7 @@ export interface DataGridPropsWithDefaultValues {
    * Select the pageSize dynamically using the component UI.
    * @default [25, 50, 100]
    */
-  pageSizeOptions: Array<number | { value: number; label: string }>;
+  pageSizeOptions: readonly (number | { value: number; label: string })[];
```

Comment:
More consistent in the codebase?
```suggestion
  pageSizeOptions: ReadonlyArray<number | { value: number; label: string }>;
```